src/Agent.Worker/TaskManager.cs

```diff
@@ -381,6 +437,154 @@ private string GetTaskZipPath(Pipelines.TaskStepDefinitionReference task)
+
+                    node6Installed = true;
+
+                    Trace.Info($"Finished getting Node 6 runner at: {nodeFolder}.");
+
+
+                }
+                catch (OperationCanceledException) when (downloadTimeout.IsCancellationRequested)
+                {
+                    Trace.Info($"Node 6 runner download has been canceled.");
+                    throw;
+                }
+                catch (SocketException ex)
+                {
+                    ExceptionsUtil.HandleSocketException(ex, downloadUrl, Trace.Warning);
+                }
+                catch (Exception ex)
+                {
+                    if (downloadTimeout.Token.IsCancellationRequested)
+                    {
+                        Trace.Warning($"Node 6 runner download has timed out after {timeoutSeconds} seconds");
+                    }
+
+                    Trace.Warning($"Failed to get package '{filePath}' from '{downloadUrl}'. Exception {ex}");
+                }
+
+                finally
+                {
+                    try
+                    {
+                        // delete .zip file
+                        if (!string.IsNullOrEmpty(filePath) && File.Exists(filePath))
+                        {
+                            Trace.Verbose("Deleting Node 6 runner package zip: {0}", filePath);
+                            IOUtil.DeleteFile(filePath);
+                        }
+                    }
+                    catch (Exception ex)
+                    {
+                        //it is not critical if we fail to delete the .zip file
+                        Trace.Warning("Failed to delete Node 6 runner package zip '{0}'. Exception: {1}", filePath, ex);
+                    }
+
+                    try
+                    {
+                        Dictionary<string, string> telemetryData = new Dictionary<string, string>
+                        {
+                            { "OS", PlatformUtil.GetSystemId() ?? "" },
+                            { "OSGroup",VarUtil.OS },
+                            { "OSArch", VarUtil.OSArchitecture},
+                            { "Node6DownloadUrl",downloadUrl},
+                            { "Node6DownloadResult",node6Installed.ToString()},
+                        };
+                        var cmd = new Command("telemetry", "publish")
+                        {
+                            Data = JsonConvert.SerializeObject(telemetryData, Formatting.None)
+                        };
+                        cmd.Properties.Add("area", "PipelinesTasks");
+                        cmd.Properties.Add("feature", "TaskManager");
+
+                        var publishTelemetryCmd = new TelemetryCommandExtension();
+                        publishTelemetryCmd.Initialize(HostContext);
+                        publishTelemetryCmd.ProcessCommand(executionContext, cmd);
+                    }
+                    catch (Exception ex)
```

Comment:
Please don't catch exceptions - it masks bugs and makes debugging more difficult.   (Please address in other places too)